### ISSUE TITLE
Cannot push port twice, or before it being pulled

### ISSUE THREAD

**Issue by wernerdegroot** [akka/alpakka-kafka#1186]
### Versions used 

`akka-stream-kafka` version 2.0.4
`akka-actor` version 2.5.30

### Expected Behavior

No exceptions

### Actual Behavior

Getting `IllegalArgumentException` once every couple of hours

### Relevant logs

```
Error in stage [akka.kafka.internal.CommitCollectorStage@48c3c78b]: Cannot push port (FlowOut(1890132076)) twice, or before it being pulled
java.lang.IllegalArgumentException: Cannot push port (FlowOut(1890132076)) twice, or before it being pulled
	at akka.stream.stage.GraphStageLogic.push(GraphStage.scala:642)
	at akka.kafka.internal.CommitCollectorStageLogic.$anonfun$consume$1(CommitCollectorStage.scala:60)
	at akka.kafka.internal.CommitCollectorStageLogic.$anonfun$consume$1$adapted(CommitCollectorStage.scala:60)
	at akka.kafka.internal.CommitCollectorStageLogic.akka$kafka$internal$CommitCollectorStageLogic$$pushDownStream(CommitCollectorStage.scala:80)
	at akka.kafka.internal.CommitCollectorStageLogic.akka$kafka$internal$CommitCollectorStageLogic$$consume(CommitCollectorStage.scala:60)
	at akka.kafka.internal.CommitCollectorStageLogic$$anon$1.onPush(CommitCollectorStage.scala:90)
	at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:523)
	at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:480)
	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:376)
	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:606)
	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:485)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:581)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:749)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$shortCircuitBatch(ActorGraphInterpreter.scala:739)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:765)
	at akka.actor.Actor.aroundReceive(Actor.scala:539)
	at akka.actor.Actor.aroundReceive$(Actor.scala:537)
	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:671)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:612)
	at akka.actor.ActorCell.invoke(ActorCell.scala:581)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
	at kamon.instrumentation.executor.ExecutorInstrumentation$InstrumentedForkJoinPool$TimingRunnable.run(ExecutorInstrumentation.scala:653)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:49)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

### Reproducible Test Case

I do not have a reproducible test case yet. I'm hoping it won't be necessary since this issue was reported before: #1165, #1159 and #1160

**Comment by seglo** [comment]
Thanks for the report. Though it's the same exception the cause is different in this case from #1159. We could take the same solution we did in #1160, but I wonder if that PR actually caused this one.

**Comment by ennru** [comment]
Your sense is correct, @seglo. Even with the `emit` introduced in #1160 unlucky timing may lead to `emit` "eating" the request which makes the subsequent `push` fail.
I have a sketch of a solution to this.

**Comment by ennru** [comment]
Hopefully fixed for good with #1186
